src/nfa_builder.rs

```diff
@@ -212,13 +212,10 @@ where
         // But, there is no problem since Daachorse does not allow an empty pattern.
         debug_assert_ne!(q[0], ROOT_STATE_ID);
 
-        // Adds a dummy output so that the output_pos is positive.
-        self.outputs.push(Output::new(V::default(), 0, None));
-
```

Comment:
Is this modification necessary? Though my understanding may be wrong, this modification enforces to do +1/-1 when setting/getting output_pos values. It seems to increase code complexity. 
Could you explain the benefit of this modification?

Reply:
This modification enforces adjusting indices but removes `V::default()`.
In my experiments, there were no significant changes in performance, so I chose the benefit of removing the Default trait bound.

Reply:
I think that removing the trait bound of Default brings a large advantage because users can use references as values as follows:
```rust
use std::collections::HashMap;

use daachorse::DoubleArrayAhoCorasick;

fn main() {
    let mut map = HashMap::new();
    map.insert("abc", vec![0, 1, 2]);
    map.insert("ab", vec![5, 6]);
    map.insert("bc", vec![4, 5]);

    let pma = DoubleArrayAhoCorasick::with_values(&map).unwrap();

    for m in pma.find_overlapping_iter("abcd") {
        dbg!(m);
    }

    print_typename(pma);
}

fn print_typename<T>(_: T) {
    println!("{}", std::any::type_name::<T>());
}
```
```
[src/main.rs:14] m = Match {
    length: 2,
    end: 2,
    value: [
        5,
        6,
    ],
}
[src/main.rs:14] m = Match {
    length: 3,
    end: 3,
    value: [
        0,
        1,
        2,
    ],
}
[src/main.rs:14] m = Match {
    length: 2,
    end: 3,
    value: [
        4,
        5,
    ],
}
daachorse::bytewise::DoubleArrayAhoCorasick<&alloc::vec::Vec<i32>>
```

Reply:
I understood. Thank you for your explanation.